app/admin/dish.rb

```diff
@@ -1,5 +1,46 @@
 ActiveAdmin.register Dish do
 
-  permit_params :name, :description, :price, :pic_url, :dish_category_id
+  permit_params :name, :description, :price, :pic_url, :dish_category_id, :restaurant_id
 
+  index do
+    selectable_column
+    id_column
+    column :name, sortable: :name
+    column :description, sortable: :description
+    column :price, sortable: :price
+    column :dish_category, sortable: :dish_category, as: :select, collection: DishCategory.all.select {|t| [t.name]}
+    column :restaurant, sortable: :restaurant
+    column :pic_url, sortable: :pic_url
+    actions
+  end
+
+  filter :name
+  filter :description
+  filter :price
+  filter :restaurant, as: :select, collection: proc{ current_admin_user.restaurants }
+
+  # filter :menus, as: :select, collection: Menu.all.each { |m| [m.name] if proc{ current_admin_user.restaurants.each { |r| r.menus.each { |rm| rm.restaurant_id == m.restaurant_id }}}}
+
+  # filter :menus, as: :select, collection: Menu.all.each { |m| [m.name] if proc{ current_admin_user.restaurants.each { |r| r.id == m.restaurant_id }}}
+  # binding.pry
+  # filter :menu_line, as: :select, collection: MenuLine.all.select{ |ml| [ml.menu_id] if proc{ current_admin_user.restaurants.each { |r| r.menus.ids.any? { |id| id == ml.menu_id} }}}
```

Comment:
Trying to make a menu filter, so you can view only dishes that belong to certain menu.